posts/2021/05/may-status-update.md

```diff
@@ -0,0 +1,101 @@
+<!--
+.. title: HPy Status Update
+.. slug: may-status-update
+.. date: 2021-05-07 10:00:00 UTC
+.. author: hodgestar
+.. tags:
+.. category:
+.. link:
+.. description:
+.. type: text
+-->
+
+
+HPy's first git commit was on Friday, 12 July 2019. Despite the crazy state
+of the world, HPy has come a long way since then. This is a quick update
+to say where we are now and what's planned for the immediate future.
+
+<!--TEASER_END-->
+
+What can one do with HPy right now?
+===================================
+
+One can already port non-trivial Python extensions to the HPy API, but
+there is also still a lot of work to do. Things one can do right now
+with HPy include:
+
+  - Initialize an extension module
+  - Create custom extension types
+  - Create custom methods and functions
+  - Interact with many built-in types (including dictionaries, tuples,
+    lists, strings, bytes and longs)
+  - Raise and check exceptions
+  - Parse arguments to C functions
+  - Call Python functions
+  - Compile an extension in either "universal" or "cpython" modes (see below)
+  - Put a "universal" extension into debug mode **at runtime** (see below
+    for how this is possible without a performance penalty).
+  - Compile extensions for POSIX and Windows (MSC).
+
+An extension written using HPy may be compiled either in "cpython" or
+"universal" mode:
+
+  - "cpython" mode produces an ordinary Python C extension that only uses
+    the HPy API internally. From the point of view of the Python implementation
+    it is an ordinary Python C extension.
+
+  - "universal" mode produces an HPy C extension. This extension does not use
+    the Python C API directly at all. A Python implementation must know how
+    to load such an extension and pass it an HPy context. Currently context
```

Comment:
First mention of "context". Could you link to the definition of this term in the docs?

Reply:
Hmm. Good point. Let me think about how to address this nicely.

Reply:
Here too my guess is that many readers with read the words "HPy Universal ABI" and have no idea what they mean.

Reply:
I changed these to avoid mentioning the context and to refer to cpython and universal ABIs.